scripts/autonomous_cross_deduplicator.py

```diff
@@ -52,38 +78,52 @@ def has_minhash_content(directory: Path) -> bool:
             and (directory / '1.minhashes').is_file())
 
 
-def collect_processed_batches(finished_batches_dir: Path) -> list[Path]:
+def assemble_targets(input_dir: Path, output_dir: Path,
+                     from_dir: Path, upto_dir):
     """
-    Collects the directories within the finished_batches_dir
-    which have date-like names and contain the following files:
-    1.doc_ids, 1.files, 1.minhashes
+    Returns a list of tuples. The tuples contain the parameters for the
+    deduplicate_other() method:
+    The input (minhash self) dir of the target
+    The list of dirs of older batches (minhash full)
+    The output (minhash full) dir for the target.
+
+    Ignores dirs which do not contain the required files
     """
-    collected_dirs = []
-    for directory in finished_batches_dir.iterdir():
-        if re.match('^[0-9_]+$', directory.name):
-            if has_minhash_content(directory):
-                collected_dirs.append(directory)
-        else:
-            logging.info(f'Directory name {directory.name} was not date-like')
-    collected_dirs.sort()
-    logging.info(f'The following directories contain fully processed '
-                 f'batches: {", ".join(str(d) for d in collected_dirs)}')
-    return collected_dirs
-
-
-def find_batch_to_process(self_dedup_dir: Path,
-                          finished_batch_names: list[str]) -> Path:
+    list_of_dirs = [dir.name for dir in sorted(input_dir.iterdir())
+                    if has_minhash_content(dir)]
+    if upto_dir:
+        upto_i = list_of_dirs.index(upto_dir)
+        list_of_dirs = list_of_dirs[:upto_i+1]
+    if from_dir:
+        from_i = list_of_dirs.index(from_dir)
+        target_list = list_of_dirs[from_i:]
+    else:
+        from_i = 0
+        target_list = list_of_dirs
+    logging.debug(f'The list of targets is: {target_list} \n They start from '
+                  f'pos {from_i} of the relevant history: {list_of_dirs}')
+    pairings = []
+    for index, target in enumerate(target_list):
+        # TODO we do not support multiple minhash files per batch.
+        target_as_input = input_dir / target / '1'
+        target_as_output = output_dir / target
+        past = [output_dir / dir / '1' for dir
+                in list_of_dirs[:from_i + index]]
+        pairings.append((target_as_input, past, target_as_output,))
+    return pairings
+
+
+def wrapped_deduplicate(input_tuple, threshold, permutations):
     """
-    Finds the directory whose name when interpreted as a date is the earliest
-    and is not contained in the blacklist of already finished batches
+    Pool does not have a function that combines the features of imap and
+    starmap, so we need a wrapper for deduplicate other. But we can't access
+    the args from here, so we still need a partial wrapped around this one...

```

Comment:
Instead of doing this, let's take https://github.com/DavidNemeskey/zim_to_corpus/blob/master/zim_to_corpus/istarmap.py and copy it over to this repo as well.

Reply:
Cool, I switched over to this one!